src/lsp_data.rs

```diff
@@ -85,11 +86,18 @@ pub mod ls_util {
         }
     }
 
-    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Range {
-        // FIXME: todo, endpos must be the end of the document, this is not correct
-
-        let end_pos = Position::new(0, 0);
-        Range{ start: Position::new(0, 0), end: end_pos }
+    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Result<Range, vfs::Error> {
```

Comment:
Given that all the clients unwrap, I think it is better if this function returns a Range and does the unwrap itself (I'm a little uneasy about exposing the vfs::Error to the rest of the RLS and I think it is easier to unwrap than to convert the error).